src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,140 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOADING_FACTOR = 0.75f;
+    private static final int RESIZE_MULTIPLIER = 2;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] bucketsArray;
+
+    public MyHashMap() {
+        threshold = (int) (DEFAULT_CAPACITY * LOADING_FACTOR);
+        bucketsArray = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newElement = new Node<>(key, value);
+        if (keyIsNew(key)) {
+            size++;
+        }
```

Comment:
can we remove this `keyIsNew(key)` private method? we actually do the similar logic twice: in `keyIsNew` and in `addElement`